src/main/scala/chisel3/tester/internal/TestEnvInterface.scala

```diff
@@ -13,6 +18,24 @@ trait TestEnvInterface {
 
   def topFileName: Option[String]
 
+  /**
+    * Will add a TargetDirAnnotation with defaultDir with "test_run_dir" path prefix to the annotations
+    * if there is not a TargetDirAnnotation already present
+    *
+    * @param defaultDir     a default directory
+    * @param annotationSeq  annotations to add it to, unless one is already there
+    * @return
+    */
+  def addDefaultTargetDir(defaultDir: String, annotationSeq: AnnotationSeq): AnnotationSeq = {
+    if(annotationSeq.exists { case _: TargetDirAnnotation => true ; case _ => false }) {
+      annotationSeq
+    }
```

Comment:
what is this `else` style and why are we using it

Reply:
reverted to your vernacular. I have been using my else style in scala since 2008, I have been using the equivalent in C since 1983. It seems to work ok :-)